pkg/console/subresource/deployment/deployment_test.go

```diff
@@ -916,95 +1077,562 @@ func Test_consoleVolumes(t *testing.T) {
+		Labels:                     labels,
+		Annotations:                map[string]string{},
+		OwnerReferences:            nil,
+		Finalizers:                 nil,
+		ClusterName:                "",
+	}
+
+	infrastructureConfigHighlyAvailable := infrastructureConfigWithTopology(configv1.HighlyAvailableTopologyMode)
+	infrastructureConfigSingleReplica := infrastructureConfigWithTopology(configv1.SingleReplicaTopologyMode)
+
+	downloadsDeploymentPodSpecSingleReplica := corev1.PodSpec{
+		NodeSelector: map[string]string{
+			"kubernetes.io/os": "linux",
+		},
+		Affinity: &corev1.Affinity{},
+		Tolerations: []corev1.Toleration{
+			{
+				Key:      "node-role.kubernetes.io/master",
+				Operator: corev1.TolerationOpExists,
+				Effect:   corev1.TaintEffectNoSchedule,
+			},
+			{
+				Key:               "node.kubernetes.io/unreachable",
+				Operator:          corev1.TolerationOpExists,
+				Effect:            corev1.TaintEffectNoExecute,
+				TolerationSeconds: &tolerationSeconds,
+			},
+			{
+				Key:               "node.kubernetes.io/not-reachable",
+				Operator:          corev1.TolerationOpExists,
+				Effect:            corev1.TaintEffectNoExecute,
+				TolerationSeconds: &tolerationSeconds,
+			},
+		},
+		SecurityContext:               &corev1.PodSecurityContext{},
+		PriorityClassName:             "system-cluster-critical",
+		TerminationGracePeriodSeconds: &gracePeriod,
+		Containers: []corev1.Container{
+			{
+				Name:                     "download-server",
+				TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+				Image:                    "",
+				ImagePullPolicy:          corev1.PullPolicy("IfNotPresent"),
+				Ports: []corev1.ContainerPort{{
+					Name:          api.DownloadsPortName,
+					Protocol:      corev1.ProtocolTCP,
+					ContainerPort: api.DownloadsPort,
+				}},
+				ReadinessProbe: &corev1.Probe{
+					Handler: corev1.Handler{
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/",
+							Port:   intstr.FromInt(api.DownloadsPort),
+							Scheme: corev1.URIScheme("HTTP"),
+						},
+					},
+					TimeoutSeconds:   1,
+					PeriodSeconds:    10,
+					SuccessThreshold: 1,
+					FailureThreshold: 3,
+				},
+				LivenessProbe: &corev1.Probe{
+					Handler: corev1.Handler{
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/",
+							Port:   intstr.FromInt(api.DownloadsPort),
+							Scheme: corev1.URIScheme("HTTP"),
+						},
+					},
+					TimeoutSeconds:   1,
+					PeriodSeconds:    10,
+					SuccessThreshold: 1,
+					FailureThreshold: 3,
+				},
+				Command: []string{"/bin/sh"},
+				Resources: corev1.ResourceRequirements{
+					Requests: map[corev1.ResourceName]resource.Quantity{
+						corev1.ResourceCPU:    resource.MustParse("10m"),
+						corev1.ResourceMemory: resource.MustParse("50Mi"),
+					},
+				},
+				Args: downloadsDeploymentTemplate.Spec.Template.Spec.Containers[0].Args,
+			},
+		},
+	}
+	downloadsDeploymentPodSpecHighAvail := downloadsDeploymentPodSpecSingleReplica
```

Comment:
not new but its a good practise 
```suggestion
	downloadsDeploymentPodSpecHighAvail := downloadsDeploymentPodSpecSingleReplica.DeepCopy()
```